ldms/src/core/ldms.c

```diff
@@ -2915,14 +2910,14 @@ void ldms_metric_array_set_double(ldms_set_t s, int mid, int idx, double v)
 void ldms_metric_array_set(ldms_set_t s, int mid, ldms_mval_t mval,
 			   size_t start, size_t count)
 {
-	ldms_mdesc_t desc = ldms_ptr_(struct ldms_value_desc, s->meta,
-				      __le32_to_cpu(s->meta->dict[mid]));
 	int i;
 	if (!s) {
```

Comment:
@baallan, It seems as if we are being inconsistent with the input parameter checking. If we are going to do this, we should do it for all of these API, not just the one that was flagged by the static analysis.  I also, think that the parameter checking in this and related cases should assert, not log a message. This is indicative of a programming error, not something that could/should be diagnosed by an admin from a log file.